### ISSUE TITLE
Error when starting kafka-1, kafka-2, kafka-4

### ISSUE THREAD

**Issue by mmacphail** [Zenika/kafka-monitoring#2]
When I run the brokers using : `docker-compose up -d kafka-1 kafka-2 kafka-3`

I get the following error:
```
kafka-1  | java.lang.IllegalArgumentException: Could not find a 'KafkaServer' or 'secure.KafkaServer' entry in the JAAS configuration. System property 'java.security.auth.login.config' is not set
kafka-1  |      at org.apache.kafka.common.security.JaasContext.defaultContext(JaasContext.java:131)
kafka-1  |      at org.apache.kafka.common.security.JaasContext.load(JaasContext.java:96)
kafka-1  |      at org.apache.kafka.common.security.JaasContext.loadServerContext(JaasContext.java:69)
kafka-1  |      at org.apache.kafka.common.network.ChannelBuilders.create(ChannelBuilders.java:143)
kafka-1  |      at org.apache.kafka.common.network.ChannelBuilders.serverChannelBuilder(ChannelBuilders.java:107)
kafka-1  |      at kafka.network.Processor.<init>(SocketServer.scala:853)
kafka-1  |      at kafka.network.SocketServer.newProcessor(SocketServer.scala:442)
kafka-1  |      at kafka.network.SocketServer.$anonfun$addDataPlaneProcessors$1(SocketServer.scala:299)
kafka-1  |      at scala.collection.immutable.Range.foreach$mVc$sp(Range.scala:190)
kafka-1  |      at kafka.network.SocketServer.addDataPlaneProcessors(SocketServer.scala:297)
kafka-1  |      at kafka.network.SocketServer.$anonfun$createDataPlaneAcceptorsAndProcessors$1(SocketServer.scala:262)
kafka-1  |      at kafka.network.SocketServer.$anonfun$createDataPlaneAcceptorsAndProcessors$1$adapted(SocketServer.scala:259)
kafka-1  |      at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:563)
kafka-1  |      at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:561)
kafka-1  |      at scala.collection.AbstractIterable.foreach(Iterable.scala:919)
kafka-1  |      at kafka.network.SocketServer.createDataPlaneAcceptorsAndProcessors(SocketServer.scala:259)
kafka-1  |      at kafka.network.SocketServer.startup(SocketServer.scala:131)
kafka-1  |      at kafka.server.KafkaServer.startup(KafkaServer.scala:285)
kafka-1  |      at kafka.Kafka$.main(Kafka.scala:109)
kafka-1  |      at kafka.Kafka.main(Kafka.scala)
```

What security config is missing ?

**Comment by mmacphail** [comment]
Current `server.properties` file :
```
inter.broker.listener.name=SECURE
super.users=User:admin
delete.topic.enable=true
listener.name.secure.scram_sha_512.sasl.jaas.config=org.apache.kafka.common.security.scram.ScramLoginModule required username=admin password=admin-secret;
auto.create.topics.enable=true
advertised.listeners=SECURE://kafka-1:9092
sasl.enabled.mechanisms=SCRAM-SHA-512
listener.security.protocol.map=SECURE:SASL_PLAINTEXT
authorizer.class.name=kafka.security.authorizer.AclAuthorizer
broker.id=101
listeners=SECURE://:9092
zookeeper.connect=zk-1:2181,zk-2:2181,zk-3:2181
confluent.topic.replication.factor=1
sasl.mechanism.inter.broker.protocol=SCRAM-SHA-512
log.dirs=/var/lib/kafka/data
offsets.topic.replication.factor=1
```

**Comment by mmacphail** [comment]
[Confluent documentation](https://docs.confluent.io/platform/current/installation/docker/config-reference.html#confluent-enterprise-ak-configuration) is wrong. It states:
```
Prefix with KAFKA_ for Apache Kafka.
Prefix with CONFLUENT_ for Confluent components.
Convert to upper-case.
Replace a period (.) with a single underscore (_).
Replace a dash (-) with double underscores (__).
Replace an underscore (_) with triple underscores (___).
```

But `KAFKA_LISTENER_NAME_SECURE_SCRAM__sha__512_sasl_jaas_config` generates `listener.name.secure.scram_sha_512.sasl.jaas.config`, which is wrong (two underscores makes a single underscore).

However I just checked and `KAFKA_LISTENER_NAME_SECURE_SCRAM___sha___512_sasl_jaas_config` generates `listener.name.secure.scram-sha-512.sasl.jaas.config`, which is correct I believe.

So replace a dash with triple underscores and an underscore with double underscore, not the other way around.

**Comment by mmacphail** [comment]
Corrected by 67a8a30
